### ISSUE TITLE
putView() sugar

### ISSUE THREAD

**Issue by nolanlawson** [pouchdb/pouchdb#2017]
@daleharvey wants to wait on introducing `putView()` into the documentation, which is a red flag to me that the code smells.  I didn't get much feedback in #1820, but I'll try again.

Basically, the problem is that the concept of design docs/views/mapreduce are too hard for the average user to understand, and it's too easy to mess up even if you're experienced with CouchDB (@neojski and I can both commiserate: https://github.com/pouchdb/mapreduce/issues/72).  So we have a few options, ranging in terms of sugar content from Raisin Bran to Pixy Stix:
#### Option 1: don't do anything (not very sugary)
- User needs to put their own damn views.  To be clear, this looks like this:

``` js
db.put({
  _id : '_design/my_view',
  views: {
    my_view :
      {
        map : function (doc) { emit(doc.name);}.toString()
      }
  }
})
```
#### Option 2: `putView()` (medium sugar content)
- Basically a simplification of putting a design doc.  The above would look like this:

``` js
db.putView('my_view', function (doc) { emit(doc.name); })
```
#### Option 3: `query(mapFun, {saveAs : 'my_view'})`
- High fructose content, attempts to save the user's query as a view `my_view` and returns an error if it can't save it for whatever reason.

My hunch is that what y'all really want is either the high-sugar version or the low-sugar version; the half-measure of `putView()` is not making anybody happy.  Or maybe something else I haven't thought of yet.  Does this sound about right?
